kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/KiePMMLTarget.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.pmml.commons.model;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import org.kie.pmml.api.enums.CAST_INTEGER;
+import org.kie.pmml.api.enums.OP_TYPE;
+import org.kie.pmml.commons.model.abstracts.AbstractKiePMMLComponent;
+
+/**
+ * @see <a href=http://dmg.org/pmml/v4-4-1/Targets.html#xsdElement_Target>Target</a>
+ */
+public class KiePMMLTarget extends AbstractKiePMMLComponent {
+
+    private static final long serialVersionUID = -6336733489238275499L;
+    private List<KiePMMLTargetValue> targetValues = null;
+    private OP_TYPE opType;
+    private String field;
+    private CAST_INTEGER castInteger;
+    private Double min = null;
+    private Double max = null;
+    private double rescaleConstant = 0;
+    private double rescaleFactor = 1;
+
+    private KiePMMLTarget(String name, List<KiePMMLExtension> extensions) {
+        super(name, extensions);
+    }
+
+    public static Builder builder(String name, List<KiePMMLExtension> extensions) {
+        return new Builder(name, extensions);
+    }
+
+    public Object modifyPrediction(Object prediction) {
+        if (!(prediction instanceof Number)) {
+            // currently unimplemented
+            return prediction;
```

Comment:
Shouldn't we rather somehow fail and throw exception if something is not implemented.

Reply:
@jiripetrlik 
I would prefer to keep going as we have done for other unimplemented/partially implemented features - mentioning the limitation but anyway providing the evaluation